### PR TITLE
Discard layer which have an extent outside the requested print extent

### PR DESCRIPTION
**Pull request by procrastinatio**
With the new WMS and WMTS import, we may now import layers which does not cover the whole Switzerland (e.g. Canton). This PR filters out these layer, which cannot be printed, _i.e._ have an extend not in the requested print extent.

[Demo](https://mf-geoadmin3.int.bgdi.ch/mom_print_layer_extent/index.html)

### CODE REVIEW

File: src/components/print/PrintDirective.js [hunk 1]
```diff
@@ -541,6 +541,19 @@ goog.require('ga_urlutils_service');
         if (!layer.visible || layer.opacity == 0) {
           return;
         }
+        // Only print layer which have an extent intersecting the print extent
+        var layerExtent = layer.getExtent();
+        if (layerExtent) {
+          var intersection = ol.extent.getIntersection(layerExtent,
```
Comment (oterral) [inline src/components/print/PrintDirective.js]: I guess you don't care to get the intersection extent, you can use this instead:

ol.extent.intersects(extent1, extent2)

Comment (oterral) [inline src/components/print/PrintDirective.js]: all in one  
```
ol.extent.intersects(layer.getExtent() || [], getPrintRectangleCoords())
```

Comment (procrastinatio) [inline src/components/print/PrintDirective.js]: No `ol.extent.intersects(extent1, extent2)`works only if `extents2` is completely enclosed in `extent1`, which is a bit too restrictive. If only a part of `extent2` is enclosed, I want it to be printed.

File: src/components/print/PrintDirective.js [hunk 2]
```diff
@@ -541,6 +541,17 @@ goog.require('ga_urlutils_service');
         if (!layer.visible || layer.opacity == 0) {
           return;
         }
+        // Only print layer which have an extent intersecting the print extent
+        if (!ol.extent.intersects(layer.getExtent() || [],
+                    getPrintRectangleCoords())) {
```
Comment (oterral) [inline src/components/print/PrintDirective.js]: indentation

**Comment by procrastinatio** [comment]
jenkins test

**Pull request merged**